### ISSUE TITLE
adapter-cloudflare 4.2.0 broke `@iconify/svelte`

### ISSUE THREAD

**Issue by katriellucas** [sveltejs/kit#12027]
> [!NOTE]
> As explained by @gtm-nayan on the comments, seems this is actually being caused by the latest Vite (5.2.X), downgrading to 5.1 also fixes this.
---
### Describe the bug

This is likely related to #12003, I can't add an Icon component from [@iconify](https://www.npmjs.com/package/@iconify/svelte) it says I have invalid html nested in the button, this is fine on v4.1.0 of the adapter. There are some other side effects, components do not render, pages do not update on change.

### Reproduction

Honestly, this bug seems to be quite elusive, and sometimes repro just don't work, not sure if it's cache/local storage in my local settings.

https://github.com/katriellucas/hydra-bug

### Logs

```Shell
Svelte SSR validation error:

<button> is invalid inside <button>

Ensure your components render valid HTML as the browser will try to repair invalid HTML, which may result in content being shifted around and will likely result in a hydration mismatch.
```

### System Info

```Shell
System:
    OS: Windows 11 10.0.22631
    CPU: (8) x64 AMD Ryzen 5 3500U with Radeon Vega Mobile Gfx
    Memory: 4.48 GB / 9.88 GB
  Binaries:
    Node: 20.10.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.2.3 - C:\Program Files\nodejs\npm.CMD
    bun: 1.0.35 - ~\.bun\bin\bun.EXE
  Browsers: {}
  npmPackages:
    @sveltejs/adapter-cloudflare: 4.2.0 => 4.2.0
    @sveltejs/kit: ^2.5.4 => 2.5.4
    @sveltejs/vite-plugin-svelte: ^3.0.2 => 3.0.2
    svelte: ^5.0.0-next.81 => 5.0.0-next.82
    vite: ^5.2.3 => 5.2.4
```

### Severity

blocking an upgrade

**Comment by gtm-nayan** [comment]
Can you downgrade you vite version to 5.1.something and check if it works or not?

I doubt the adapter version would affect this in anyway, what's more likely is your vite version got updated at the same time you updated your adapter and that's the source of the problem.

**Comment by katriellucas** [comment]
> Can you downgrade your vite version to 5.1.something and check if it works or not?
> 
> I doubt the adapter version would affect this in anyway, what's more likely is your vite version got updated at the same time you updated your adapter and that's the source of the problem.

You are right, it's weird that if you just downgrade the other package, it seems to works fine, I will also add a note then.

**Comment by gtm-nayan** [comment]
@katriellucas if vite 5.1.x fixed it for you then the bug should've been fixed in 5.2.5, please upgrade to that version and report back.

**Comment by katriellucas** [comment]
@gtm-nayan just reporting that the latest vite version is fine, I will be closing the issue.
